### ISSUE TITLE
Refreshing Jellyfin-Vue while logged in breaks in Firefox

### ISSUE THREAD

**Issue by JVT038** [jellyfin/jellyfin-vue#2269]
### Description of the bug

Jellyfin-Vue breaks when you reload the page on Firefox while logged in.
The splash screen shows the Jellyfin logo and that's it. Nothing loads anymore.

### Steps to reproduce

1. Log in while using Firefox
2. Refresh the page (F5 or CTRL + R)
3. See error 

### Expected behavior

I expect Jellyfin-Vue to function properly after reloading the page 

### Screenshots

I'm stuck on this splash screen:
![afbeelding](https://github.com/jellyfin/jellyfin-vue/assets/47184046/6e6acd75-3632-4049-9be6-cfe681e07026)

### Platform

Windows

### Browser

Firefox

### Jellyfin server version

10.8.13

### Additional context

```text
When I use Chrome instead, this error does not occur.

This also occurs in troubleshoot mode, which is a 'safe mode' in Firefox that disables all extensions. So it's not caused by extensions either.
```

**Comment by EffakT** [comment]
I noticed this as-well, seems something is breaking in regards to the auth in localStorage, by deleting the "auth" key in localStorage, this allows you back in (You have to re-login though). Rolling back to unstable.2024-03-14.22615a1 works, so likely something that changed between that version and this version.

**Comment by ferferga** [comment]
@EffakT @JVT038 Do you guys see any call to the logout endpoint in the Network tab? Or any with a response error code of 401?

**Comment by EffakT** [comment]
@ferferga  No errors. I'm guessing its related to either the middlewares or network changes 3 days ago.

**Comment by ferferga** [comment]
No, those are purely cosmetic and the meta handling is no different besides we now reuse the reactive object instead of recreating it every time.

I was also experimenting the same issue in my end because of a race condition caused by the ``flush: 'sync'`` watchers, where:
* They performed an authenticated request to server before the token was populated
* After the request was dispatched, the token and axios were properly configured
* The response with 401 lands from the server.
* Given there's an authentication failure, the token is cleared rightaway (we can introduce a verification request to know if this is actually true, which is my proposed solution, but keep reading).

See https://github.com/jellyfin/jellyfin-vue/blob/master/frontend/src/plugins/remote/axios/index.ts#L25

There are no errors at all since the deauth it's expected to work this way and be handled gracefully without errors in console (just the snackbar is going to be displayed to the user telling the session has expired). However, the client gets stuck in an endless loading state because everything happened so fast with the sync watchers that the session gets revoked even before the current user is set [and this watcher triggers](https://github.com/jellyfin/jellyfin-vue/blob/master/frontend/src/plugins/router/index.ts#L88)

There's also no fallback for Suspense pages, since all the offline state handling is currently WIP and provided in a best effort basis ([see here](https://github.com/jellyfin/jellyfin-vue/pull/2201)), those kind of situations will be handled more gracefully in the future.

That's why I need a good trace of the network requests between the client and the server, because console message are not useful at all in this case. If the logout endpoint is called, I can fix this in a couple of seconds. If not, this is going to be much complex.

**Comment by EffakT** [comment]
I've attached the .har file, does that help? I have it authenticated against the Jellyfin Demo server.

[9ffcaf85.jf-vue.pages.dev_Archive [24-03-21 11-49-38].har.zip](https://github.com/jellyfin/jellyfin-vue/files/14674837/9ffcaf85.jf-vue.pages.dev_Archive.24-03-21.11-49-38.har.zip)

**Comment by ferferga** [comment]
@EffakT Thank you very much for the HAR file. please verify if the issue is solved in the linked PR.

cc @JVT038

**Comment by EffakT** [comment]
@ferferga Still occurring on my end.
[33cb1cc4.jf-vue.pages.dev_Archive [24-03-22 07-44-59].har.zip](https://github.com/jellyfin/jellyfin-vue/files/14705692/33cb1cc4.jf-vue.pages.dev_Archive.24-03-22.07-44-59.har.zip)

**Comment by JVT038** [comment]
It's still occuring for me too
